### DISCUSSION TITLE
Migrating from Cordova to Capacitor in a yarn workspace

### DISCUSSION THREAD

**Discussion by ryandthorn** [ionic-team/capacitor#3461] [Q&A] [answered]
I'm currently trying to migrate a Cordova app over to Capacitor following Ionic's [documented instructions](https://capacitorjs.com/docs/cordova/migrating-from-cordova-to-capacitor). We're excited to move to Capacitor, but I'm running into an issue right off the bat trying to install `@capacitor/cli` and `@capacitor/core`. 

We use a monorepo configured with yarn workspaces, so we maintain our packages in a single package.json and corresponding yarn.lock at the root level, and our apps live in subdirectories. Trying to install the CLI/core packages at the project level throws an error that a dependency cannot be located. Installing these packages at the root level seems to work, but I then run into a similar error with `yarn cap add android` where `cap` doesn't seem to find the dependencies in the workspace root folder from the project folder.

Has anyone run into a similar issue with yarn workspaces and resolved it? The docs say that Capacitor is compatible with yarn, but my experience (as well as [this merged PR](https://github.com/ionic-team/capacitor/pull/3409) referring to yarn dependency installation and monorepo support as "scope creep") makes me think otherwise and may force us to move on to another option instead of Capacitor.

**Comment by imhoffd** [comment, marked as answer]
Just to be clear, that PR removes the **invocations** of yarn from the Capacitor CLI. In Capacitor 3, we won't be invoking npm/yarn/whatever, which will _improve_ compatibility. The scope creep was in reference to us having to support the plethora of project structures, which is a lot easier to do if we don't assume where you want dependencies installed.

Can you elaborate on the errors you're seeing?

**Reply by ryandthorn** [reply]
Thank you for the response; being package-manager neutral makes a lot of sense. Here's what I've been seeing at each step:

`yarn add @capacitor/core @capacitor/cli`

> error An unexpected error occurred: "expected workspace package to exist for \"eslint-plugin-jsx-a11y\""

Going up a level to the root directory (where our modules/package/lock are) and adding core and CLI using the -W flag to add the packages to the workspace root seems to install the packages successfully.

I can then `cd` back into the app and run `yarn cap init`. Then I run `yarn cap add android` and get the same missing package error as before:

> error An unexpected error occurred: "expected workspace package to exist for \"eslint-plugin-jsx-a11y\""

Also, `npx cap add android` results in:

> command not found: cap

Please let me know if you'd like any more info.

**Reply by imhoffd** [reply]
`npx cap` resulting in "command not found" makes sense. npx doesn't know how to work with yarn workspaces.

I would be surprised if the error about `eslint-plugin-jsx-a11y` was related to Capacitor. Can you try using `yarn add` on other packages and see if you get the same error (making sure your project is reset to a working state before Capacitor was added)?

**Reply by ryandthorn** [reply]
Good suggestion. I got the same error trying to add an unrelated package, so it does indeed seem like it's not related to Capacitor. I'll try to debug this on my end, and I'll follow up when I'm able to get this up and running. Thanks again for your time.

**Reply by ryandthorn** [reply]
Sorry for the late response. We're up and running in our yarn workspace by installing Capacitor packages to the workspace root. Happy to have migrated successfully to Capacitor.

**Reply by imhoffd** [reply]
Thanks for following up! Great to hear 🍻
